### PR TITLE
Add ServiceDeclarationSymbol

### LINKED ISSUE #27493: Service Symbol is not visible in Semantic Model
**Description:**
Cannot retrieve compiler API Service Symbol details from the Semantic Model.

**Usecase:** List services details for the project packages API in LS. (Related PR: https://github.com/ballerina-platform/ballerina-lang/pull/27488) 

**Steps to reproduce:**
`module.getCompilation().getSemanticModel()` cannot be used to retrieve Service and Resource details.

**Affected Versions:**
SLP 8

**OS, DB, other environment details and versions:**

**Related Issues (optional):**

**Suggested Labels (optional):**

**Suggested Assignees (optional):**

### PR DESCRIPTION
**Pull request by pubudu91**
## Purpose
This PR introduces a symbol for service declarations. At a high level, this PR adds/makes the following changes:
- Makes use of the `BServiceSymbol` as the symbol of service decls. This symbol holds additional info that aren't included in the symbol of the class generated for the service decl. 
- Adds a separate top level BIR node for service decls and persist service symbol info in the BIR. This means the BIR version gets bumped as well.
- Introduce a public symbol for service decls. This also means required and related util APIs have also been added to support resource functions and resource paths.
- Adds the service decls to BIR spec and tests for it.

Fixes #27493

## Approach
The `ServiceDeclarationSymbol` is modeled as follows:

service-symbol-Page-1 (3)

And the `ResourcePath` as follows:

service-symbol-Page-3

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: language-server/modules/langserver-core/src/test/java/org/ballerinalang/langserver/completion/CompletionTestNew.java [hunk 1]
```diff
@@ -80,13 +80,13 @@ public void test(String config, String configPath) throws WorkspaceDocumentExcep
...
-//            java.nio.file.Files.write(FileUtils.RES_DIR.resolve(configJsonPath),
-//                                      obj.toString().getBytes(java.nio.charset.StandardCharsets.UTF_8));
-//
+            JsonObject obj = new JsonObject();
+            obj.add("position", configJsonObject.get("position"));
+            obj.add("source", configJsonObject.get("source"));
+            obj.add("items", resultList);
+            java.nio.file.Files.write(FileUtils.RES_DIR.resolve(configJsonPath),
+                                      obj.toString().getBytes(java.nio.charset.StandardCharsets.UTF_8));
+
```
Comment (nadeeshaan) [inline language-server/modules/langserver-core/src/test/java/org/ballerinalang/langserver/completion/CompletionTestNew.java:89]: We need to comment this

**Review by nadeeshaan — CHANGES_REQUESTED** [review]

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/writer/BIRBinaryWriter.java [hunk 2]
```diff
@@ -54,7 +54,8 @@
  *
  * @since 0.980.0
  */
-public class BIRBinaryWriter {
+public class
+BIRBinaryWriter {
```
Comment (Kishanthan) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/writer/BIRBinaryWriter.java:58]: why did we break this line?

Comment (pubudu91) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/writer/BIRBinaryWriter.java:58]: looks like this was by mistake 😬 will revert

**Review by nadeeshaan — DISMISSED** [review]
LGTM

**Review by rdhananjaya — APPROVED** [review]

**Pull request merged**